Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+﻿using RimWorld;
+using SaveOurShip2;
+using System.Text;
+using Verse;
+
+namespace TheCafFiend
+{
+    [StaticConstructorOnStartup]
+    public class CompSpinalEngineTrail : CompEngineTrail
+    {
+        public override string CompInspectStringExtra()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (fullyFormed)
+            {
+                stringBuilder.Append($"Fuel burn rate per second: {cachedFuelUse} of {cachedFuelAllowed} supported");
+                stringBuilder.AppendInNewLine($"Thrust: {cachedThrust * 500}");
+            }
+            else
+            {
+                stringBuilder.Append(cachedError.Colorize(UnityEngine.Color.red));
+            }
+            return stringBuilder.ToString();
+        }
+        public virtual CompProperties_SpinalEngineTrail Properties //Wait is this dirty I just want to be clear it's not hiding hrm
+        {
+            get { return props as CompProperties_SpinalEngineTrail; }
+        }
+        public class ReturnValue
+        {
+            public float thrustAmp;
+            public float fuelAmp;
+            public float fuelAllowAmp;
+            public float powerUseAmp;
+            public int supportWeight;
+            public string playerError;
+            public bool fullyFormed;
+        }
+        private int cachedThrust = 0;
```

Comment:
🤔  basically to represent that this is a cached value on the controller of the structure, because it can be impacted by other parts: Thus the getters checking if the multiblock has been complete and just returning that as opposed to re-calculating the entire structure multiple times a second during combat. Possibly premature optimization, mind, but I was already interacting with a cache anyhow so it made sense in my head?

Reply:
I wouldn't say it's a premature optimization if you know you don't need to recalculate it all the time.  If for example you planned on adding some sort of alien race that made your engines more efficient based on being assigned to them then maybe you'd want something to recalculate the engine thrust - BUT I wouldn't have the calculation here, that's just an example.

At the end of the day, if the name makes sense to you keep it ;)  You're leaving this code for future you to read and understand.

For me it's a semantic intent of the variable.  Cached in my head is "I took this and I'm saving it for a later calculation", but "thrust" or "currentThrust" is "this is the value I have right now", which is still subject to change if I have parts of the multiblock replaced / shot off.

Reply:
Huh, yep, and here I would invalidate my cache if a bit was shot off (in fact, I do!)